src/overlays/actors/ovl_En_Tanron4/z_en_tanron4.c

```diff
@@ -28,22 +44,241 @@ const ActorInit En_Tanron4_InitVars = {
+            this->timeInfluence = 1500.0f;
+            thisx->world.pos.y += 1500.0f;
+        }
+    }
+}
+
+void EnTanron4_Destroy(Actor* thisx, GlobalContext* globalCtx) {
+}
+
+void EnTanron4_SetupFlyNearHome(EnTanron4* this) {
+    this->actionFunc = EnTanron4_FlyNearHome;
+    Animation_MorphToLoop(&this->skelAnime, &gSeagullFlapAnim, 0.0f);
+}
+
+void EnTanron4_FlyNearHome(EnTanron4* this, GlobalContext* globalCtx) {
+    f32 xDiff;
+    f32 yDiff;
+    f32 zDiff;
+    f32 distToTarget;
+    s16 xRot;
+    s16 yRot;
+    s16 zRot;
+
+    // `timeInfluence` controls both the height of the seagulls and when they are visible.
+    // They fly higher in the sky as the night goes on, and they dissapear as dawn approaches.
+    if ((gSaveContext.time > CLOCK_TIME(20, 0)) || (gSaveContext.time < CLOCK_TIME(4, 0))) {
+        Math_ApproachF(&this->timeInfluence, 1500.0f, 1.0f, 1.0f);
+    } else {
+        Math_ApproachZeroF(&this->timeInfluence, 1.0f, 1.0f);
+    }
+
+    xDiff = this->targetPos.x - this->actor.world.pos.x;
+    yDiff = (this->targetPos.y + this->timeInfluence) - this->actor.world.pos.y;
+    zDiff = this->targetPos.z - this->actor.world.pos.z;
+
+    distToTarget = sqrtf(SQ(xDiff) + SQ(zDiff));
+
+    if ((this->timers[SEAGULL_TIMER_CHOOSE_TARGET] == 0) || (distToTarget < 100.0f)) {
+        this->targetPos.x = this->actor.home.pos.x + randPlusMinusPoint5Scaled(500.0f);
+        this->targetPos.y = this->actor.home.pos.y + randPlusMinusPoint5Scaled(100.0f);
+        this->targetPos.z = this->actor.home.pos.z + randPlusMinusPoint5Scaled(500.0f);
+
+        this->timers[SEAGULL_TIMER_CHOOSE_TARGET] = Rand_ZeroFloat(100.0f) + 60.0f;
+        this->step = 0;
+    }
+
+    yRot = Math_Atan2S(xDiff, zDiff);
+    xRot = Math_Atan2S(yDiff, distToTarget);
+    zRot = Math_SmoothStepToS(&this->actor.world.rot.y, yRot, 0xA, this->step, 0);
+
+    if (zRot > 0x1000) {
+        zRot = 0x1000;
+    } else if (zRot < -0x1000) {
+        zRot = -0x1000;
+    }
+
+    Math_ApproachS(&this->actor.world.rot.x, xRot, 0xA, this->step);
+    Math_ApproachS(&this->actor.world.rot.z, -zRot, 0xA, this->step);
+
+    Math_ApproachS(&this->step, 0x200, 1, 0x10);
+
+    SkelAnime_Update(&this->skelAnime);
+
+    switch (this->flyState) {
+        case SEAGULL_FLY_FLAP:
+            if ((this->timers[SEAGULL_TIMER_FLY_STATE] == 0) &&
+                (Animation_OnFrame(&this->skelAnime, 2.0f + KREG(42)))) {
+                this->flyState = SEAGULL_FLY_GLIDE;
+                this->timers[SEAGULL_TIMER_FLY_STATE] = Rand_ZeroFloat(50.0f) + 50.0f;
+                Animation_MorphToLoop(&this->skelAnime, &gSeagullFlapAnim, -15.0f + KREG(43));
+                this->skelAnime.curFrame = 2.0f + KREG(42);
+                this->skelAnime.playSpeed = 0.0f;
+            }
+            break;
+
+        case SEAGULL_FLY_GLIDE:
+            if (((this->randRollTimer & 7) == 0) && (Rand_ZeroOne() < 0.5f)) {
```

Comment:
Would this match with a `% 8` instead?
```suggestion
            if (((this->randRollTimer % 8) == 0) && (Rand_ZeroOne() < 0.5f)) {
```